test/test_spectral_ops.py

```diff
@@ -1125,10 +1125,21 @@ def _test_istft_is_inverse_of_stft(stft_kwargs):
                 for i in range(num_trials):
                     original = torch.randn(*sizes, dtype=dtype, device=device)
                     stft = torch.stft(original, return_complex=True, **stft_kwargs)
-                    inversed = torch.istft(stft, length=original.size(1), **istft_kwargs)
-
+                    inversed = torch.istft(stft, length=original.size(-1), **istft_kwargs)
+                    n_frames = stft.size(-1)
+                    if stft_kwargs["center"] is True:
+                        len_expected = stft_kwargs["n_fft"] // 2 + stft_kwargs["hop_length"] * (n_frames - 1)
+                    else:
+                        len_expected = stft_kwargs["n_fft"] + stft_kwargs["hop_length"] * (n_frames - 1)
                     # trim the original for case when constructed signal is shorter than original
-                    original = original[..., :inversed.size(-1)]
+                    if original.size(-1) > len_expected:
```

Comment:
nit: I think you can always perform the below operations, as even if `original.size(-1) <= len_expected`, the below operations should still work as PyTorch support tensors with 0 in its dimensions

Reply:
That's true. I think the conditional logic is more intuitive, as there are cases when the ``len_expected`` is equal to the original waveform's length. We can keep that for reading.